google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/RejectMerchantCenterLink.java

```diff
@@ -0,0 +1,160 @@
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID of the Google Ads account to reject the link request.
+   * @param merchantCenterAccountId the Merchant Center account ID for the account requesting to
+   *     link.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, long merchantCenterAccountId) {
+
+    // Reject a pending link request or unlink an enabled link for a Google Ads account with
+    // customerId from a Merchant Center account with merchantCenterAccountId.
+    try (MerchantCenterLinkServiceClient merchantCenterLinkService =
+        googleAdsClient.getLatestVersion().createMerchantCenterLinkServiceClient()) {
+      ListMerchantCenterLinksResponse response =
+          merchantCenterLinkService.listMerchantCenterLinks(
+              ListMerchantCenterLinksRequest.newBuilder()
+                  .setCustomerId(Long.toString(customerId))
+                  .build());
+
+      System.out.printf(
+          "%d Merchant Center link(s) found with the following resource name(s):%n",
+          response.getMerchantCenterLinksCount());
+
+      for (MerchantCenterLink merchantCenterLink : response.getMerchantCenterLinksList()) {
+        System.out.printf("'%s'%n", merchantCenterLink.getResourceName());
+
+        // Check if there is a link for the Merchant Center account we are looking for.
+        if (merchantCenterAccountId == merchantCenterLink.getId().getValue()) {
+          // If the Merchant Center link is pending, reject it by removing the link.
+          // If the Merchant Center link is enabled, unlink Merchant Center from Google Ads by
+          // removing the link.
+          // In both cases, the remove action is the same.
+          removeMerchantCenterLink(merchantCenterLinkService, customerId, merchantCenterLink);
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   * Remove a Merchant Center link from a Google Ads client customer account.
+   *
+   * @param merchantCenterLinkServiceClient the MerchantCenterLinkService client.
+   * @param customerId the client customer ID of the Google Ads account that has the link request.
+   * @param merchantCenterLink the MerchantCenterLink object to remove.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void removeMerchantCenterLink(
+      MerchantCenterLinkServiceClient merchantCenterLinkServiceClient,
+      long customerId,
+      MerchantCenterLink merchantCenterLink) {
```

Comment:
There is no reason to pass the full MerchantCenterLink object in this example because there is no field mask comparison being performed. We can just pass the resource name.

Reply:
It's true, this isn't required (very perceptive of you!) but I think passing the object makes the code more self-documenting. I can change this if you feel strongly about it.

Reply:
I think it's OK to pass the whole object, since the method and its calling context are all dealing with the `MerchantCenterLink` object itself as well. In the comments on the operation builder, it points out that it's specifying the resource name, which I think makes it pretty clear that's all you need.